src/EndpointManager.sol

```diff
@@ -25,6 +25,17 @@ abstract contract EndpointManager is IEndpointManager, OwnableUpgradeable, Reent
         BURNING
     }
 
+    // @dev Information about attestations for a given message.
+    struct AttestationInfo {
+        // whether this message has been executed
+        bool executed;
+        // bitmap of endpoints that have attested to this message (NOTE: might contain disabled endpoints)
+        uint64 attestedEndpoints;
+    }
+
+    // Maps are keyed by hash of EndpointManagerMessage.
+    mapping(bytes32 => AttestationInfo) public managerMessageAttestations;
```

Comment:
The `EndpointManager` doesn't use `AttestationInfo.attestedEndpoints` -- I'm wondering whether the replay protection in the `EndpointManager` should be a `mapping(bytes32 => bool)` to just track which digests have been executed, and then the standalone contract could implement multi-sig threshold logic on top of this by keeping another storage variable to track `mapping(bytes32 => uint64) digestAttestedEndpoints`.

Agreed we need better naming for these contracts. `EnpointManagerStandalone` should really be named `EndpointManager`, and `EndpointManager` should be named something like `BaseAbstractMessenger` since it's really becoming responsible for `sendMessage`, `quoteDeliveryPrice`, and `_executeMsg` (these methods will also become external once we pull out the business logic for transferring tokens into a separate contract).

Reply:
Yeah, this is a little annoying, and separating the two would definitely be the cleanest. It would however add an extra 20k gas overhead on each message in the threshold case (since the endpoint attestations and the replay protection would now be stored in different slots).

Reply:
Ok, we can also revisit this later once we complete more refactoring and think on whether this trade-off is worth it.